build-info-extractor/src/main/java/org/jfrog/build/extractor/executor/CommandExecutor.java

```diff
@@ -85,64 +84,91 @@ static String getFixedWindowsPath(String path) {
-        Process process = null;
         ExecutorService service = Executors.newFixedThreadPool(2);
         try {
             CommandResults commandRes = new CommandResults();
-            process = runProcess(execDir, args, env, logger);
-            StreamReader inputStreamReader = new StreamReader(process.getInputStream());
-            StreamReader errorStreamReader = new StreamReader(process.getErrorStream());
-            service.submit(inputStreamReader);
-            service.submit(errorStreamReader);
-            process.waitFor();
-            service.shutdown();
-            service.awaitTermination(10, TimeUnit.SECONDS);
-            commandRes.setRes(inputStreamReader.getOutput());
-            commandRes.setErr(errorStreamReader.getOutput());
-            commandRes.setExitValue(process.exitValue());
-            return commandRes;
+            Process process = runProcess(execDir, args, credentials, env, logger);
+            // The output stream is not necessary in non-interactive scenarios
+            process.getOutputStream().close();
+            try (InputStream inputStream = process.getInputStream();
+                 InputStream errorStream = process.getErrorStream()) {
+                StreamReader inputStreamReader = new StreamReader(inputStream);
+                StreamReader errorStreamReader = new StreamReader(errorStream);
+                service.submit(inputStreamReader);
+                service.submit(errorStreamReader);
+                process.waitFor();
+                service.shutdown();
+                boolean terminatedProperly = service.awaitTermination(10, TimeUnit.SECONDS);
+                if (!terminatedProperly && logger != null) {
+                    logger.error("Process '" + String.join(" ", args) + "' had been terminated forcibly after timeout of 10 seconds.");
+                }
+                commandRes.setRes(inputStreamReader.getOutput());
+                commandRes.setErr(errorStreamReader.getOutput());
+                commandRes.setExitValue(process.exitValue());
+                return commandRes;
+            }
         } finally {
-            closeStreams(process);
             service.shutdownNow();
         }
     }
 
-    private static void closeStreams(Process process) {
-        if (process != null) {
-            IOUtils.closeQuietly(process.getInputStream());
-            IOUtils.closeQuietly(process.getOutputStream());
-            IOUtils.closeQuietly(process.getErrorStream());
+    private static Process runProcess(File execDir, List<String> args, List<String> credentials, String[] env, Log logger) throws IOException {
+        if (credentials != null) {
+            args.addAll(credentials);
         }
+        if (SystemUtils.IS_OS_WINDOWS) {
+            args.addAll(0, Arrays.asList("cmd", "/c"));
+        } else if (SystemUtils.IS_OS_MAC) {
+            // In MacOS, the arguments for '/bin/sh -c' must be a single string. For example "/bin/sh","-c", "npm i".
```

Comment:
Shouldn't "must" be changed to "must not" here?

Reply:
No. That is the main difference. Arguments in MacOS are a single string.